### PR TITLE
feat: Wrap List in `ul` for accessibility

### LINKED ISSUE #15884: List component should have option to render as ol/ul and each item becomes wrapped in li
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
A List of objects is generally best semantically marked up as an `<ol>` or `<ul>`

Improved semantics improves SEO & screen reader experience.  For example, when coming across an `<ol>` with 5 `<li>` inside the screen reader will say "List of 5 items".  This is very helpful to screen readers.

Currently the component just outputs a lot of `<div>` which isn't semantically great.

Further, if you try and use just straight `<ul>`/`<ol>` in antd, it is very unsatisfying to have to style it yourself, and hope you are sticking to Ant Design principles.

### What does the proposed API look like?
Maybe `htmlType` = ?"ol" | "ul".  If not provided, render as it currently does

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

close #15884 

### 💡 Solution

Wrap List.Item under `ul`.

### 📝 Changelog

Wrap List.Item under `ul` to enhance accessibility.

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/list/Item.tsx [hunk 1]
```diff
@@ -112,9 +112,10 @@ export default class Item extends React.Component<ListItemProps, any> {
         ))}
       </ul>
     );
+    const Tag = grid ? 'div' : 'li';
     const itemChildren = (
-      <div
-        {...others}
+      <Tag
+        {...others as any} // `li` element don't have `onCopy` prop. Cast to any.
```
Comment (ztplz) [inline components/list/Item.tsx]: `onCopy` 是标准dom属性 怎么会没有呢？ 被我们占用了？

**Comment by Stephen2** [comment]
This is amazing!  Thank you for considering doing this so fast.

I think it's important to allow developer to choose between: `ol`, `ul` and even just as it currently is (`div`s)

Each has different semantic meaning

Amazing that you jumped on this idea so quickly, thank you from all my heart 🙏

File: components/list/Item.tsx (hunk 1 above)
Comment (zombieJ) [reply components/list/Item.tsx]: 一个是 `ClipboardEvent<HTMLLIElement>` ，另一个一个是 `ClipboardEvent<HTMLDivElement>`

**Comment by zombieJ** [comment]
Hi @Stephen2, I've considered this. Since current list we don't provider the number display in List. I prefer `ul` better than `ol` cause the List component not provides order id in UI and also it supports `pagination` which means `ol` also not the real index.

File: components/list/Item.tsx (hunk 1 above)
Comment (ztplz) [reply components/list/Item.tsx]: 用的时候直接React. ClipboardEventHandler就行了 总比any好啊

**Comment by Stephen2** [comment]
@zombieJ it's important to separate the semantic meaning of `Ordered List` versus the visual appearance.

If the list is ordered in any way, it's more accurate to use `ol`.  Ignore the fact that it tries to add numbers as dot points - just `list-style: none` and this problem is solved.

Here is a fantastic page of information, and why it's important to be semantically correct, for Accessibility WCAG purposes:
https://www.w3.org/TR/WCAG20-TECHS/H48.html

**Comment by Stephen2** [comment]
I am glad we can get `ul` soon, and hope you will consider optional `ol` as well.  Both have their purpose, and CSS can be used to style them in identical way, as desired.

File: components/list/Item.tsx (hunk 1 above)
Comment (zombieJ) [reply components/list/Item.tsx]: 这个对外不暴露，只是把 div 的 props 传给 div | li。其中 div 的 onCopy 是 ClipboardEvent<HTMLDivElement>，而 li 需要的是 ClipboardEvent<HTMLLIElement>。两者不一样所以报错，这边根据 Tag 是 div 还是 li 转成对应的 props 没什么必要。

Comment (ztplz) [reply components/list/Item.tsx]: 你 omit 掉onCopy 再自己加个 ClipboardEvent 属性

Comment (zombieJ) [reply components/list/Item.tsx]: 如果之后 TS 又改了其他方法又得再 omit 再加，为了类型正确而正确没啥必要的。

**Comment by afc163** [comment]
Conflicting

File: components/comment/__tests__/__snapshots__/demo.test.js.snap [hunk 2]
```diff
@@ -216,108 +216,112 @@ exports[`renders ./components/comment/demo/list.md correctly 1`] = `
...
       class="ant-spin-container"
     >
-      <div
-        class="ant-comment"
+      <ul
+        class="ant-list-items"
       >
         <div
-          class="ant-comment-inner"
+          class="ant-comment"
```
Comment (afc163) [inline components/comment/__tests__/__snapshots__/demo.test.js.snap]: ul 下是不是应该跟 li

File: components/comment/demo/list.md [hunk 3]
```diff
@@ -53,13 +53,15 @@ ReactDOM.render(
...
     renderItem={item => (
-      <Comment
-        actions={item.actions}
-        author={item.author}
-        avatar={item.avatar}
-        content={item.content}
-        datetime={item.datetime}
-      />
+      <li>
+        <Comment
```
Comment (afc163) [inline components/comment/demo/list.md]: 这也挺蛋疼的，不如 `<Comment component="li" />`?

Comment (zombieJ) [reply components/comment/demo/list.md]: 为了结构正确而加一个属性，不是特别有必要。
其实按理说，应该是这边遵循我们 List 的 Demo，renderItem 返回的第一层应该是 List.Item。但是它带了横线的样式所以没用它包。用 li 代替 List.Item 看着不太舒服，但是相对正确一些。

**Review by afc163 — APPROVED** [review]

**Comment by nurgasemetey** [comment]
I can't find this in documentation. How can I find information on this?

**Pull request merged**